Dockerfile

```diff
@@ -4,17 +4,22 @@ ENV TZ=America/Chicago
 RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 
 RUN apt upgrade -y
-RUN apt update && apt install -y sudo wget gnupg2 git gcc gfortran libboost-dev bzip2 openmpi-bin flex build-essential bison libboost-all-dev vim libsqlite3-dev numactl sqlite3 gdb
-RUN wget -q -O - https://repo.radeon.com/rocm/rocm.gpg.key | sudo apt-key add -
-RUN echo 'deb [arch=amd64] https://repo.radeon.com/rocm/apt/debian/ ubuntu main' | sudo tee /etc/apt/sources.list.d/rocm.list
-RUN apt update
-RUN apt install -y rocm-dev6.0.0 rocm-libs6.0.0 build-essential libssl-dev swig numpy-stl
+RUN apt update && apt install -y sudo wget gnupg2 git gcc gfortran libboost-dev bzip2 openmpi-bin flex build-essential bison libboost-all-dev vim libsqlite3-dev numactl sqlite3 gdb libgtest-dev
+
+#Rocm 6.1.1
+RUN mkdir --parents --mode=0755 /etc/apt/keyrings
+RUN wget https://repo.radeon.com/rocm/rocm.gpg.key -O - | gpg --dearmor | sudo tee /etc/apt/keyrings/rocm.gpg > /dev/null
+RUN echo 'deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/6.1.1 jammy main' | sudo tee /etc/apt/sources.list.d/rocm.list
+RUN apt update && apt install -y rocm-dev6.1.1 rocm-libs6.1.1
+
+RUN apt update && apt install -y build-essential libssl-dev swig python3
+RUN pip install pytest scipy numpy==1.23.5
 
 COPY target.lst /opt/rocm/bin/
 
 # recent cmake required for FAISS 
 WORKDIR /root
-RUN wget https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1.tar.gz && tar -zxvf cmake-3.23.1.tar.gz && cd cmake-3.23.1 && ./bootstrap && make -j && make install
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.29.4/cmake-3.29.4.tar.gz && tar -zxvf cmake-3.29.4.tar.gz && cd cmake-3.29.4 && ./bootstrap && make -j && make install
```

Comment:
why update? I deliberately kept it at the minimal supported version.

Reply:
Updated when testing other stuff. Not needed. Will revert

Reply:
3.23.1 no longer works. Must be 3.24.0+. Fixed: e18bd4c176800c0ce3c246d380bdb788e2fb94a0